x/axelarnet/message_handler.go

```diff
@@ -398,3 +404,12 @@ func deductFee(ctx sdk.Context, b types.BankKeeper, fee *Fee, token keeper.Coin,
 
 	return token, b.SendCoins(ctx, types.AxelarGMPAccount, recipient, sdk.NewCoins(coin))
 }
+
+// validateReceiver rejects uppercase GMP account address
+func validateReceiver(receiver string) error {
+	if strings.ToUpper(receiver) == receiver && types.AxelarGMPAccount.Equals(funcs.Must(sdk.AccAddressFromBech32(receiver))) {
```

Comment:
Could also check if the ToLower is equal but not exactly equal to the GMP account

Reply:
bech32 rejects mix case string